### PR TITLE
FCE-811/use room service plugin

### PR DESCRIPTION
**Pull request by czerwiukk**
## Description

Tidy up the code structure and use fastify plugin to inject logic to routes.

## Motivation and Context

We need to have a proper example on how to integrate Fishjam with modern libraries.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)

### CODE REVIEW

File: examples/room-manager/src/plugins/fishjam.ts [hunk 1]
```diff
@@ -1,24 +1,35 @@
+import fp from 'fastify-plugin';
```
Comment (MiloszFilimowski) [inline examples/room-manager/src/plugins/fishjam.ts:1]: **nitpick:** This shortcut might be a bit misleading, it's often used to indicate functional programming API, for example, "lodash/fp" etc. 

And another thing is it clashes with fishjam. Is this `fastify-plugin` or `fishjam-plugin`.

I'd suggest to just import it as `fastifyPlugin`.

File: examples/room-manager/src/plugins/fishjam.ts [hunk 2]
```diff
@@ -31,7 +42,7 @@ export class RoomService {
 
     if (!peer) {
       fastify.log.info({ name: 'Creating peer' });
-      return await this.createPeer(roomName, username);
+      return await createPeer(roomName, username);
```
Comment (MiloszFilimowski) [inline examples/room-manager/src/plugins/fishjam.ts:45]: **nitpick:** `await` is redundant in this context.

File: examples/room-manager/src/routes/rooms.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      const url = httpToWebsocket(fastify.config.FISHJAM_URL);
+
+      // When creating a URL object from a URL without a path (e.g., `http://localhost:5002`),
+      // the `href` field may contain an additional '/' at the end (`http://localhost:5002/`).
+      const urlWithoutTrailingSlash = removeTrailingSlash(url);
+
+      return { ...accessData, url: urlWithoutTrailingSlash };
+    } catch (error: unknown) {
+      const [parsedError, errorCode] = parseError(error);
+      return res.status(errorCode).send(parsedError.detail);
```
Comment (MiloszFilimowski) [inline examples/room-manager/src/routes/rooms.ts:24]: **question:** Is this return needed?

File: examples/room-manager/src/routes/rooms.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  const getRoomAccessHandler = async (roomName: string, peerName: string, res: FastifyReply) => {
+    try {
+      const accessData = await fastify.fishjam.getPeerAccess(roomName, peerName);
+      const url = httpToWebsocket(fastify.config.FISHJAM_URL);
+
+      // When creating a URL object from a URL without a path (e.g., `http://localhost:5002`),
+      // the `href` field may contain an additional '/' at the end (`http://localhost:5002/`).
+      const urlWithoutTrailingSlash = removeTrailingSlash(url);
+
+      return { ...accessData, url: urlWithoutTrailingSlash };
```
Comment (MiloszFilimowski) [inline examples/room-manager/src/routes/rooms.ts:21]: **question:** Does it work without `res.send`? Is there some kind of middleware?

File: examples/room-manager/src/plugins/fishjam.ts (hunk 1 above)
Comment (czerwiukk) [reply examples/room-manager/src/plugins/fishjam.ts:1]: this is as well the default approach in fastify documentation, so I think fastify users will not be confused

File: examples/room-manager/src/routes/rooms.ts (hunk 4 above)
Comment (czerwiukk) [reply examples/room-manager/src/routes/rooms.ts:21]: yes

File: examples/room-manager/src/plugins/fishjam.ts (hunk 2 above)
Comment (czerwiukk) [reply examples/room-manager/src/plugins/fishjam.ts:45]: out of scope, but true, fixed

**Review by PiotrWodecki — APPROVED** [review]
Tested manually ✅

**Review by Karolk99 — APPROVED** [review]

**Pull request merged**